### PR TITLE
combine english and chinese, remove nan

### PR DESCRIPTION
**Pull request by MargarettMao**
## Description

This contains simple modification in the perplexity test. 

This version combines 'en'(stands for English) column and 'ch'(stands for Chinese) column into one column as 'result'. This modification removes the 'nan' in the test perplexity test result.

### CODE REVIEW

**Comment by MargarettMao** [comment]
4790bab9-65a4-45ec-a674-abfcb760b6d0
96ec6b3c-3dbb-4c06-adb9-47d2f5003d22

**Comment by hxsz1997** [comment]
Please also modify [ppl_csv_to_html.py](https://github.com/intel-analytics/ipex-llm/blob/main/python/llm/test/benchmark/perplexity/ppl_csv_to_html.py) and [fp16.csv](https://github.com/intel-analytics/ipex-llm/blob/main/python/llm/test/benchmark/perplexity/fp16.csv)

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 1]
```diff
@@ -37,8 +37,8 @@ def nonzero_min(lst):
     non_zero_lst = [num for num in lst if num > 0.0]
     return min(non_zero_lst) if non_zero_lst else None
 
-def is_diffs_within_normal_range(diff_en, diff_zh, threshold=5.0):
-    return not any(diff < (-threshold) for diff in diff_en + diff_zh if isinstance(diff, float))
+def is_diffs_within_normal_range(diff_ppl_result, threshold=5.0):
+    return not any(diff < (-threshold) for diff in diff_ppl_result + diff_ppl_result if isinstance(diff, float))#diff_en + diff_zh
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:41]: `for diff in diff_ppl_result` not `for diff in diff_ppl_result + diff_ppl_result`

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 2]
```diff
@@ -150,40 +167,48 @@ def main():
...
+
             if not in_previous_flag:
-                last_en[latest_csv_ind]=pd.NA
-                diff_en[latest_csv_ind]=pd.NA
-                last_zh[latest_csv_ind]=pd.NA
-                diff_zh[latest_csv_ind]=pd.NA
+                last_ppl_result[latest_csv_ind]=pd.NA
+                diff_ppl_result[latest_csv_ind]=pd.NA
+                # last_zh[latest_csv_ind]=pd.NA
+                # diff_zh[latest_csv_ind]=pd.NA
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:190]: delete

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 3]
```diff
@@ -104,42 +110,53 @@ def main():
...
 
             latest_csv_model=latest_csv_row['Model'].strip()
             latest_csv_precision=latest_csv_row['Precision'].strip()
-            latest_en=latest_csv_row[en]
-            latest_zh=latest_csv_row[zh]
+            latest_ppl_result=latest_csv_row[ppl_result]#latest_en=latest_csv_row[en]
+            print("This is the latest_ppl_result",latest_ppl_result)
+            # latest_zh=latest_csv_row[zh]
 
             in_previous_flag=False
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:161]: delete

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 4]
```diff
@@ -104,42 +110,53 @@ def main():
 
     diffs_within_normal_range = True
 
-    # Add display of FP16 values for each model and add percentage difference column
-    for task in ['en', 'zh']:
+        # Add display of FP16 values for each model and add percentage difference column
+
+    for task in ['ppl_result']:#['en', 'zh']:
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:115]: delete

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 5]
```diff
@@ -104,42 +110,53 @@ def main():
...
         latest_csv[f'{task}_FP16'] = latest_csv['Model'].apply(lambda model: fp16_dict.get(model, {}).get(task, 'N/A'))
+        # print("This is the stuff I want to check",latest_csv['Model'].apply(lambda model: fp16_dict.get(model, {}).get(task, 'N/A') == 'N/A'))
         latest_csv[f'{task}_diff_FP16(%)'] = latest_csv.apply(lambda row: calculate_percentage_difference(row[task], row[f'{task}_FP16']), axis=1)
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:118]: `task` change to `'ppl_result'`

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 6]
```diff
@@ -150,40 +167,48 @@ def main():
...
-
+        columns={'ppl_result': '{:.2f}', 'last_ppl_result': '{:.2f}', 'last_diff_ppl_result(%)': '{:.2f}'}
+                #{'en': '{:.2f}', 'zh': '{:.2f}', 'last_en': '{:.2f}', 'diff_en(%)': '{:.2f}',
+                #'last_zh': '{:.2f}', 'diff_zh(%)': '{:.2f}'}
         latest_csv.drop('Index', axis=1, inplace=True)
 
         styled_df = latest_csv.style.format(columns).applymap(lambda val: highlight_vals(val, max=3.0, is_last=True), subset=subset1)
-        for task in ['en', 'zh']:
+        for task in ['ppl_result']:#['en', 'zh']:
             styled_df = styled_df.applymap(lambda val: highlight_vals(val, max=highlight_threshold, is_last=False), subset=[f'{task}_diff_FP16(%)'])
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:209]: delete `for task in ['ppl_result']:#['en', 'zh']:` and change `task` to `'ppl_result'`

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 7]
```diff
@@ -150,40 +167,48 @@ def main():
...
+                diff_ppl_result[latest_csv_ind]=pd.NA
+                # last_zh[latest_csv_ind]=pd.NA
+                # diff_zh[latest_csv_ind]=pd.NA
 
-        latest_csv.insert(loc=9,column='last_en',value=last_en)
-        latest_csv.insert(loc=10,column='diff_en(%)',value=diff_en)
-        latest_csv.insert(loc=11,column='last_zh',value=last_zh)
-        latest_csv.insert(loc=12,column='diff_zh(%)',value=diff_zh)
+        latest_csv.insert(loc=6,column='last_ppl_result',value=last_ppl_result)
+        latest_csv.insert(loc=7,column='last_diff_ppl_result(%)',value=diff_ppl_result)
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:193]: change `'last_diff_ppl_result(%)'` to `'ppl_result_diff_last(%)'`

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 8]
```diff
@@ -195,7 +220,26 @@ def main():
         with open(daily_html, 'w') as f:
             f.write(html_output)
     else:
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:222]: There is duplication with the previous code.

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 9]
```diff
@@ -58,11 +58,11 @@ def create_fp16_dict(fp16_path):
     fp16_dict = {}
     for _, row in fp16_df.iterrows():
         model = row['Model']
-        # Formalize the data to have 2 decimal places
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:61]: Not delete.

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 10]
```diff
@@ -104,42 +104,42 @@ def main():
...
-    # Add display of FP16 values for each model and add percentage difference column
-    for task in ['en', 'zh']:
-        latest_csv[f'{task}_FP16'] = latest_csv['Model'].apply(lambda model: fp16_dict.get(model, {}).get(task, 'N/A'))
-        latest_csv[f'{task}_diff_FP16(%)'] = latest_csv.apply(lambda row: calculate_percentage_difference(row[task], row[f'{task}_FP16']), axis=1)
+        # Add display of FP16 values for each model and add percentage difference column
+
+    task = 'ppl_result'
+    latest_csv[f'{task}_FP16'] = latest_csv['Model'].apply(lambda model: fp16_dict.get(model, {}).get(task, 'N/A'))
+    latest_csv[f'{task}_diff_FP16(%)'] = latest_csv.apply(lambda row: calculate_percentage_difference(row[task], row[f'{task}_FP16']), axis=1)
+    print(csv_files)
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:112]: Delete `task = 'ppl_result'`. Replace `task` to `'ppl_result'`

File: python/llm/test/benchmark/perplexity/ppl_csv_to_html.py [hunk 11]
```diff
@@ -150,57 +150,67 @@ def main():
...
-        styled_df = latest_csv.style.format(columns).applymap(lambda val: highlight_vals(val, max=3.0, is_last=True), subset=subset1)
-        for task in ['en', 'zh']:
-            styled_df = styled_df.applymap(lambda val: highlight_vals(val, max=highlight_threshold, is_last=False), subset=[f'{task}_diff_FP16(%)'])
-        
-        # add css style to restrict width and wrap text
-        styled_df.set_table_styles([{
-            'selector': 'th, td',
-            'props': [('max-width', '88px'), ('word-wrap', 'break-word')]
-        }], overwrite=False)
+        subset1='ppl_result_diff_last(%)'
```
Comment (hxsz1997) [inline python/llm/test/benchmark/perplexity/ppl_csv_to_html.py:172]: Delete `subset1='ppl_result_diff_last(%)'`

**Review by hxsz1997 — APPROVED** [review]
LTGM

**Pull request merged**